Feature/part 4

SergioFah opened the pull request.

SergioFah committed:
- implementação para consumir backend

SergioFah committed:
no message

SergioFah committed:
- fix api endpoints

SergioFah committed:
Update README.md

SergioFah committed:
Merge branch 'feature/spring-consume' into develop

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/controller/ProductPageController.java:
```diff
@@ -16,12 +18,17 @@
-	private String selectedLine;
 
-	private ProductDAO productDAO;
+	private Long selectedLineId;
+
+	private Service service;
+
+	@FXML
+	private Label serverStatus;
 
 	Image loading;
```
By our team code convention, it's necessary to declare the `@FXML` attributes first.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/controller/ProductPageController.java:
```diff
@@ -45,53 +52,70 @@ public class ProductPageController {
-		productDAO = new ProductDAO();
-		populateComboBox();
+		this.service = new Service();
+		if(service.getServerStatus()) {
```
Turn `if(` into `if (`

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/controller/ProductPageController.java:
```diff
@@ -45,53 +52,70 @@ public class ProductPageController {
+		if(service.getServerStatus()) {
+			this.lineList = service.getLines();
+			populateComboBox();
+		}else{
```
turn `}else{` into `} else {`

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/controller/ProductPageController.java:
```diff
@@ -45,53 +52,70 @@ public class ProductPageController {
+		}
 	}
 
 	public void populateComboBox(){
```
The left brace must always be separated from the previous character by most Java code conventions and our convention.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/controller/ProductPageController.java:
```diff
@@ -45,53 +52,70 @@ public class ProductPageController {
 	private void populateTreeView() {
-		
-    	List<Category> categoryList = productDAO.getCategoriesFromLine(selectedLine);
 
```
Remove line 92

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/controller/ProductPageController.java:
```diff
@@ -45,53 +52,70 @@ public class ProductPageController {
+				populateModelDetails(selectedProduct);
             }
         });
 	}
```
Remove line 128

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/model/Category.java:
```diff
@@ -0,0 +1,59 @@
+        "line",
+        "category"
+})
+public class Category {
```
Turn into DTO class

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/model/Category.java:
```diff
@@ -0,0 +1,59 @@
+})
+public class Category {
+
+    @JsonProperty("id")
```
`@JsonProperty` is not necessary here. Please, remove it and do the same thing for the other unnecessary properties.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/model/Line.java:
```diff
@@ -0,0 +1,42 @@
+        "id",
+        "line"
+})
+public class Line {
```
Turn into DTO class

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/model/Line.java:
```diff
@@ -0,0 +1,42 @@
+})
+public class Line {
+
+    @JsonProperty("id")
```
`@JsonProperty` is not necessary here. Please, remove it and do the same thing for the other unnecessary properties.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/model/Product.java:
```diff
@@ -0,0 +1,98 @@
+})
+public class Product {
+
+    @JsonProperty("id")
```
`@JsonProperty` is not necessary here. Please, remove it and do the same thing for the other unnecessary properties.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/model/Product.java:
```diff
@@ -0,0 +1,98 @@
+        "description",
+        "imageUrl"
+})
+public class Product {
```
Turn into DTO class

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/service/Service.java:
```diff
@@ -0,0 +1,69 @@
+import java.util.List;
+import java.util.Optional;
+
+public class Service {
```
In the same thought used in backend, it's necessary to Split this classe into 3 services classes: CategoryService, LineService and ProductService.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/MainApp.java:
```diff
@@ -1,30 +1,43 @@
 		} catch (IOException e) {
 			e.printStackTrace();
 		}
+
```
Remove line 36

Adriel4raujo reviewed (changes requested):
Please, check the pendencies highlighted in the comments.

SergioFah committed:
### Removed
- unnecessary json tags
### Changed
- service layer separation
- DTO renamed
- minor conventions adjustments

SergioFah committed:
### Removed
- unnecessary json tags
### Changed
- service layer separation
- DTO renamed
- minor conventions adjustments

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/service/CategoryService.java:
```diff
@@ -0,0 +1,18 @@
+
+public class CategoryService {
+    public List<CategoryDTO> getCategoriesFromLineId(Long id) {
+        Service service = new Service();
```
The  `service` attribute mus be a class global object, since class may will have more methods that would use the attribute.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/service/LineService.java:
```diff
@@ -0,0 +1,18 @@
+
+public class LineService {
+    public List<LineDTO> getLines() {
+        Service service = new Service();
```
The  `service` attribute mus be a class global object, since class may will have more methods that would use the attribute.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/service/ProductService.java:
```diff
@@ -0,0 +1,29 @@
+public class ProductService {
+
+    public List<ProductDTO> getProductsFromCategoryId(Long id) {
+        Service service = new Service();
```
The  `service` attribute mus be a class global object, since class have others methods that use it.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/service/ProductService.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    public ProductDTO getProductFromId(Long id) {
+        Service service = new Service();
```
The  `service` attribute mus be a class global object, since class have others methods that use it.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/service/Service.java:
```diff
@@ -0,0 +1,29 @@
+
+import org.springframework.web.reactive.function.client.WebClient;
+
+public class Service {
+    private WebClient webClient;
+    public Service() {
+        this.webClient = WebClient
+                .builder()
+                .baseUrl("http://localhost:8080")
+                .build();
+    }
+    public WebClient getWebClient() {
+        return webClient;
+    }
```
It's necessary to separate class declaration, attribute declaration, constructor declaration and first method declaration.

Adriel4raujo commented on src/main/java/com/sergiofah/integracao/controller/ProductPageController.java:
```diff
@@ -46,57 +50,81 @@ public class ProductPageController {
-
-        TreeItem<String> rootItem = new TreeItem<>(selectedLine);
+		CategoryService categoryService = new CategoryService();
+		ProductService productService = new ProductService();
```
All services objects could be global attributos but productService must be because it is used not only in `populateTreeView()` method but in `selectionTreeViewHandler()` as well.

Adriel4raujo reviewed (changes requested):
Some additional details must be improved. Please, verify them in the comments.

SergioFah committed:
### Changed
- services to global scope on controller
- minor conventions adjustments

SergioFah committed:
### Fixed
- exit tomcat when closing javafx stage
### Improved
- global services on each service class

SergioFah committed:
### Changed
- minor conventions adjustments

Adriel4raujo reviewed (approved):
Approved!
